### PR TITLE
[Spark dataset runner] Add direct translation of Combine.GroupedValues

### PR DESCRIPTION
**Pull request by mosche**
Adding direct translation of Combine.GroupedValues.
This is a minor improvement to TPC-DS query 83, but by no means the issue for the big difference (related to #23845).

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by mosche** [comment]
R: @aromanenko-dev

**Comment by mosche** [comment]
Run Spark StructuredStreaming ValidatesRunner

**Comment by mosche** [comment]
Kind ping @aromanenko-dev

**Review by aromanenko-dev — APPROVED** [review]
thanks, LGTM, just a couple of notes

File: runners/spark/3/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/TransformTranslator.java [hunk 1]
```diff
@@ -69,6 +69,10 @@ public final void translate(
     translate(transform, new Context(appliedTransform, cxt));
   }
 
+  public boolean canTranslate(TransformT transform) {
```
[the remark this answers is not in the document]
Comment (mosche) [reply runners/spark/3/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/TransformTranslator.java:79]: done 👍

**Comment by aromanenko-dev** [comment]
Thanks! 
One thing that we missed before merging - could you add a unit test for this new translator and its case?

**Comment by mosche** [comment]
Thanks @aromanenko-dev 🎉

**Pull request merged**